app/models/topic.rb

```diff
@@ -1,4 +1,4 @@
 class Topic < ActiveRecord::Base
   has_many :posts, dependent: :destroy
-  has_many :sponsoredpost, dependent: :destroy
+  has_many :sponsored_posts, dependent: :destroy
```

Comment:
This is just one of those dumb conventions that you learn by doing it wrong a bunch of times. :)